ecosystem/typescript/sdk/src/aptos_client.ts

```diff
@@ -206,59 +212,51 @@ export class AptosClient {
-      gas_unit_price: "1",
-      // Unix timestamp, in seconds + 10 seconds
-      expiration_timestamp_secs: (Math.floor(Date.now() / 1000) + 10).toString(),
-      payload,
-      ...(options || {}),
-    };
-  }
+  ): Promise<TxnBuilderTypes.RawTransaction> {
+    if (payload.type !== "entry_function_payload") {
+      throw new Error("Only entry_function_payload is supported.");
+    }
 
-  /**
-   * Converts a transaction request by `generate_transaction` into it's binary hex BCS representation, ready for
-   * signing and submitting.
-   * Generally you may want to use `signTransaction`, as it takes care of this step + signing
-   * @param txnRequest A raw transaction generated by `generateTransaction` method
-   * @returns A hex-encoded string prefixed with `0x` and fulfilled with two hex digits per byte
-   */
-  async createSigningMessage(request: Gen.EncodeSubmissionRequest): Promise<string> {
-    return this.client.transactions.encodeSubmission(request);
+    const entryFunctionPayload = payload as Gen.TransactionPayload_EntryFunctionPayload;
+
+    const config: RemoteABIBuilderConfig = { sender };
+    if (options?.sequence_number) {
+      config.sequenceNumber = options.sequence_number;
+    }
+
+    if (options?.gas_unit_price) {
+      config.gasUnitPrice = options.gas_unit_price;
+    }
+
+    if (options?.max_gas_amount) {
+      config.maxGasAmount = options.max_gas_amount;
+    }
+
+    if (options?.expiration_timestamp_secs) {
+      const timestamp = Number.parseInt(options.expiration_timestamp_secs, 10);
+      config.expSecFromNow = timestamp - Math.floor(Date.now() / 1000);
+    }
+
+    const builder = new TransactionBuilderRemoteABI(this, config);
+    return builder.build(
+      entryFunctionPayload.function,
+      entryFunctionPayload.type_arguments,
+      entryFunctionPayload.arguments,
+    );
   }
 
   /** Converts a transaction request produced by `generateTransaction` into a properly
    * signed transaction, which can then be submitted to the blockchain
    * @param accountFrom AptosAccount of transaction sender
-   * @param txnRequest A raw transaction generated by `generateTransaction` method
+   * @param rawTransaction A raw transaction generated by `generateTransaction` method
    * @returns A transaction, signed with sender account
    */
+  // eslint-disable-next-line class-methods-use-this
   async signTransaction(
     accountFrom: AptosAccount,
-    txnRequest: Gen.SubmitTransactionRequest,
-  ): Promise<Gen.SubmitTransactionRequest> {
-    const message = await this.createSigningMessage(txnRequest);
-    const signatureHex = accountFrom.signHexString(message.substring(2));
-
-    const transactionSignature: Gen.TransactionSignature = {
-      type: "ed25519_signature",
-      public_key: accountFrom.pubKey().hex(),
-      signature: signatureHex.hex(),
-    };
-
-    return { ...txnRequest, signature: transactionSignature };
+    rawTransaction: TxnBuilderTypes.RawTransaction,
```

Comment:
Let's make sure to clearly announce this is a breaking change. Could you add something to the changelog as part of this PR, just in the unreleased section?

Reply:
yep.